common/src/main/java/org/jboss/gm/common/Configuration.java

```diff
@@ -224,20 +228,23 @@ public DependencyPrecedence convert(Method method, String input) {
 
     /**
      * Return the current configuration as a formatted string.
+     *
+     * @return the current configuration as a formatted string
      */
     default String dumpCurrentConfig() {
-
-        StringBuilder currentProperties = new StringBuilder();
-        for (Method method : Configuration.class.getMethods()) {
-            if (method.isAnnotationPresent(Config.Key.class)) {
-                Config.Key val = method.getAnnotation(Config.Key.class);
-                currentProperties.append(System.lineSeparator());
-                currentProperties.append('\t');
-                currentProperties.append(val.value());
-                currentProperties.append(" : ");
-                currentProperties.append(this.getProperties().get(val.value()));
-            }
+        final List<String> values = Arrays.stream(Configuration.class.getMethods())
+                .map(method -> method.getAnnotation(Key.class)).filter(Objects::nonNull).map(Key::value).sorted()
+                .collect(Collectors.toList());
+        final StringBuilder currentProperties = new StringBuilder();
+
+        for (String value : values) {
+            currentProperties.append('\t');
+            currentProperties.append(value);
+            currentProperties.append(" : ");
+            currentProperties.append(this.getProperties().get(value));
+            currentProperties.append(System.lineSeparator());
```

Comment:
Actually no ; this now means it looks like 
```
[AlignmentTask.java:118] Configuration now has properties      AProxDeployUrl : null
```
instead of
```
[AlignmentTask.java:110] Configuration now has properties 
        dependencySource : REST
```

Reply:
I am thinking the newline really belongs in the log statements themselves. What do you think?

```
logger.info("Configuration now has properties {}{}", System.lineSeparator(),
                configuration.dumpCurrentConfig());
```

If you want to keep it in the method, I think I'd still leave the newline at the end of the loop and add an additional one that's always printed regardless of whether it loops at the beginning. This seems more correct to me.

Reply:
I just think the formatting now is not like it was before regardless of the internal implementation

Reply:
I put an initial newline there, and I added a `regex` pattern to the test, so you can see there as well.

Reply:
The test doesn't seem to be passing now?

Reply:
It was a problem with the regex syntax which should test:

```
<NL>
(<KEY> : <VALUE> <NL>)*
```